wgpu-core/src/command/render.rs

```diff
@@ -1187,10 +1211,21 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                         .buffers
                         .use_extend(&*buffer_guard, buffer_id, (), BufferUse::INDIRECT)
                         .unwrap();
-                    assert!(buffer.usage.contains(BufferUsage::INDIRECT), "An invalid drawIndirect call has been made. The buffer usage is {:?} which does not contain required usage INDIRECT", buffer.usage);
+                    assert!(buffer.usage.contains(BufferUsage::INDIRECT), "An invalid drawIndirect call has been made. The indirect buffer usage is {:?} which does not contain required usage INDIRECT", buffer.usage);
+
+                    let begin_offset = offset;
+                    let end_offset = offset + 16;
+                    assert!(
+                        end_offset <= buffer.size,
+                        "drawIndirect with offset {} uses bytes {}..{} which overruns indirect buffer of size {}",
+                        offset,
+                        begin_offset,
+                        end_offset,
+                        buffer.size
+                    );
 
                     unsafe {
-                        raw.draw_indirect(&buffer.raw, offset, 1, 0);
+                        raw.draw_indirect(&buffer.raw, offset, 1, 16);
```

Comment:
is this required by D3D12 only? If Vulkan allows 0, and d3d12 requires the exact stride, it sounds to me that we should fix this at gfx-rs backend level. Would you be interested in helping with this? I haven't yet published the latest update you made, so we can squeeze this little fix in.

Reply:
Sure thing. This is just an assert in the D3D12 backend, so I'll just swap that out with a `if stride == 0 { stride = 16 }` or so.